lib/MqttClient.js

```diff
@@ -136,10 +155,12 @@ class MqttClient {
     connect() {
         if (!this.client || (this.client && this.client.connected === false && this.client.reconnecting === false)) {
             const options = {};
-            if (this.caPath) {
-                options.ca = fs.readFileSync(this.caPath);
-            }
-            this.client = mqtt.connect(this.brokerURL, options);
+            if (this.clientId) options.clientId = this.clientId;
+            if (this.username) options.username = this.username;
+            if (this.password) options.password = this.password;
+            if (this.caPath) options.ca = fs.readFileSync(this.caPath);
+
+            this.client = mqtt.connect((this.usetls ? "mqtts://" : "mqtt://") + this.server + ":" + this.port, options);
```

Comment:
(nitpick) wrap this line

Reply:
How long should be the maximum length?
The if statement 7 lines above is also that long.

Maybe we should configure that in eslint.

Reply:
fixed that line